ONSdigital/dp-frontend-renderer
# Feature/healthcheck

### What

- Converted to Go module
- Updated CI to use go 1.13.8, and updated /health endpoint in nomad plan
- Added new healthcheck library
- Makefile now generates always the same file to prevent duplicates, and it is ignored by git (will be generated as part of the build)
- Implemented graceful shutdown with timeout

### How to review

- Make sure code changes make sense
- Make sure unit tests pass
- You can try the /health endpoint (`GET http://localhost:20010/health`)
- Check that graceful shutdown works - you can optionally add a time.Sleep(*time.Second) in the shutdown goroutine, to trigger the timeout.
- In an shell without having exported GOPATH, try to run `make build` and `make run`. Validate that a file assets/data.go is generated, with encoded data in the first case and links to asset files in the second case.

### Who can review

Anyone.
